### DISCUSSION TITLE
<title>Problem with rate limiting and API access

### DISCUSSION THREAD

**Discussion by ghost** [community/community#70154] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am using a third-party app that uses GitHub API and am getting bit by GitHub rate limiting. The documentation suggests that I should supply a personal access token. This applies to an organization. I have been attempting to do this for over a week and have failed. I get completely confused and lost within the documentation offered that just seems to lead me nowhere. How can I get to know what to do? In particular, I am confused by the difference between ¨classic¨ and "fine-grained" tokens and what I should use. All I want to do is to circumvent the rate-limiting problem.

**Comment by seyLu** [comment]
Hi @jgossage,

According to the docs, these are the rate limits depending on the type of user accessing the resource:
<table>
<tr>
  <th>User Type</th> 
  <th>Allowed Requests per hour</th>
</tr>
<tr>
  <td>Unauthenticated User</td>
  <td>60</td>
</tr>
<tr>
  <td>Workflow using <code>GITHUB_TOKEN</code></td>
  <td>1000</td>
</tr>
<tr>
  <td>User authenticated with <code>PAT</code></td>
  <td>5000</td>
</tr>
<tr>
  <td>Github/Oauth App owned by Github Enterprise Cloud org</td>
  <td>15 000</td>
</tr>
</>
</table>

Classic and Fine-grained tokens are just Personal Access Tokens -- the difference being that fine-grained allows more granular control over token permissions and is generally more secure than classic tokens.

You can read more at:
- [Github docs: REST API rate limiting](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting)
- [Github docs: Types of PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#types-of-personal-access-tokens)
